src/main/java/com/softserve/demo/service/impl/EmailServiceImpl.java

```diff
@@ -32,6 +36,16 @@
      * Provided by thymeleaf.
      */
     private final TemplateEngine templateEngine;
+    /**
+     * Address of our host to be inserted in email messages.
+     */
+    @Value("${hostlink}")
+    private String HOSTLINK;
+    /**
+     * This executor service helps us to not wait for smtp server to response
+     * and just rely on separate thread to do this.
+     */
+    private ExecutorService singleThreadExecutor;
```

Comment:
What about @Async?
https://spring.io/guides/gs/async-method/

Reply:
Yeah! I read about this annotation and it is shorter to write. But I'm afraid about mechanism behind the scenes. It wants my class to be subclassable. Did you use it? I will change it if it is safe.

Reply:
It is used on my current project, and i do not think that something from spring was not tested before adding to release. Anyway, if you'll have problems later, you can easily change everything back.

Reply:
OK! Thank you!